query-engine/connector-test-kit/src/test/scala/util/TestServer.scala

```diff
@@ -46,127 +25,29 @@ case class TestServer() extends PlayJsonExtensions {
       errorCount: Int = 1,
       errorContains: String = ""
   ): JsValue = {
-    val result = awaitInfinitely {
-      querySchemaAsync(
-        query = query,
+    val result =
+      queryBinaryCLI(
+        query = query.stripMargin,
         project = project,
       )
-    }
 
     // Ignore error codes for external tests (0) and containment checks ("")
     result.assertFailingResponse(0, errorCount, "")
     result
   }
 
-  private def querySchemaAsync(
-      query: String,
-      project: Project
-  ): Future[JsValue] = {
-    val (port, queryEngineProcess) = startQueryEngine(project)
-
-    println(s"query engine started on port $port")
-    println(s"Query: $query")
-
-    Future {
-      queryPrismaProcess(query, port)
-    }.map(r => r.jsonBody.get)
-      .transform { r =>
-        println(s"Query result: $r")
-        queryEngineProcess.destroyForcibly().waitFor()
-        r
-      }
-  }
-
-  val nextPort = new AtomicInteger(4000)
-
-  private def startQueryEngine(project: Project): (Int, java.lang.Process) = {
-    import java.lang.ProcessBuilder.Redirect
-
-    // TODO: discuss with Dom whether we want to keep the legacy mode
-    val pb         = new java.lang.ProcessBuilder(EnvVars.prismaBinaryPath, "--legacy")
-    val workingDir = new java.io.File(".")
-
-    val fullDataModel = project.dataModelWithDataSourceConfig
-    // Important: Rust requires UTF-8 encoding (encodeToString uses Latin-1)
-    val encoded = Base64.getEncoder.encode(fullDataModel.getBytes(StandardCharsets.UTF_8))
-    val envVar  = new String(encoded, StandardCharsets.UTF_8)
-    val port    = nextPort.incrementAndGet()
-
-    pb.environment.put("PRISMA_DML", envVar)
-    pb.environment.put("PORT", port.toString)
-    pb.environment.put("PRISMA_LOG_QUERIES", sys.env.getOrElse("PRISMA_LOG_QUERIES", "n"))
-    pb.environment.put("RUST_LOG", sys.env.getOrElse("RUST_LOG", "info"))
-
-    pb.directory(workingDir)
-    pb.redirectErrorStream(true)
-    pb.redirectOutput(Redirect.INHERIT)
-
-    val process = pb.start
-
-    waitUntilServerIsUp(port)
-
-    (port, process)
-  }
-
-  private def queryPrismaProcess(query: String, port: Int): QueryEngineResponse = {
-    val url = new URL(s"http://127.0.0.1:$port")
-    val con = url.openConnection().asInstanceOf[HttpURLConnection]
+  def queryBinaryCLI(query: String, project: Project) = {
+    import sys.process._
 
-    con.setDoOutput(true)
-    con.setRequestMethod("POST")
-    con.setRequestProperty("Content-Type", "application/json")
+    val formattedQuery = query.stripMargin.replace("\n", "")
+    val encoded        = Base64.getEncoder.encode(formattedQuery.getBytes(StandardCharsets.UTF_8))
+    val encoded_query  = new String(encoded, StandardCharsets.UTF_8)
```

Comment:
Those 2 lines are the same code as in the other file. I think it would be good to centralize the encoding and decoding in a central singleton object. Maybe call it `UTF8Base64`?